store/bench_test.go

```diff
@@ -0,0 +1,289 @@
+package store
+
+import (
+	crand "crypto/rand"
+	"encoding/binary"
+	"fmt"
+	"math"
+	"math/rand"
+	"os"
+	"path/filepath"
+	"sort"
+	"testing"
+
+	"github.com/cosmos/iavl"
+	"github.com/stretchr/testify/require"
+
+	"github.com/cosmos/cosmos-sdk/db/badgerdb"
+	"github.com/cosmos/cosmos-sdk/db/rocksdb"
+	storev1 "github.com/cosmos/cosmos-sdk/store/iavl"
+	"github.com/cosmos/cosmos-sdk/store/types"
+	storev2 "github.com/cosmos/cosmos-sdk/store/v2/flat"
+	dbm "github.com/tendermint/tm-db"
+)
+
+var (
+	cacheSize = 100
+)
+
+func randBytes(numBytes int) []byte {
+	b := make([]byte, numBytes)
+	_, _ = crand.Read(b)
+	return b
+}
+
+type percentages struct {
+	has    uint
+	get    uint
+	set    uint
+	delete uint
+}
+
+type counts struct {
+	has    int
+	get    int
+	set    int
+	delete int
+}
+
+func generateSampledPercentages() []*percentages {
+	var sampledPercentages []*percentages
+	sampleX := &percentages{has: 5, get: 65, set: 30, delete: 0}
```

Comment:
for a reference benchmark let's use the following:
```suggestion
	sampleX := &percentages{has: 2, get: 55, set: 40, delete: 3}
```